main.cpp

```diff
@@ -58,49 +56,68 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const char* n);   //1
+    int value;
+    std::string name;
 };
 
-struct <#structName1#>                                //4
+T::T(int v, const char* n) : value(v), name(n) {}
+
+struct Compare                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if(a != nullptr)
+        {
+            if( a->value < b->value ) return a;
+        }
+        if(b != nullptr)
+        {
+            if( a->value > b->value ) return b;
+        }
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float waypoint1 { 0 }, waypoint2 { 0 };
+    float reduceDistance(float* newValue)      //12
     {
-        
+        std::cout << "U's waypoint1 value: " << this->waypoint1 << std::endl;
+        if(newValue != nullptr)
+        {
+            this->waypoint1 = *newValue;
+        }
+        std::cout << "U's waypoint1 updated value: " << this->waypoint1 << std::endl;
+        while( std::abs(this->waypoint2 - this->waypoint1) > 0.001f )
+        {           
+            this->waypoint2 += 0.001f ;
+        }
+        std::cout << "U's waypoint2 updated value: " << this->waypoint2 << std::endl;
+        return this->waypoint2 * this->waypoint1;
```

Comment:
You need to correct the nullptr checks in the `static` function below before this function is revised to have the same internal logic.
- [ ] revise